pages/submit.js

```diff
@@ -473,9 +487,13 @@ class SubmitDapp extends React.Component {
     if (twitterHandle && twitterHandle.includes('@')) {
       twitterHandle = twitterHandle.replace('@', '')
     }
+
+    const { referralCode, refSource } = this.state
```

Comment:
We actually probs don't even need to save this stuff to state, and instead do something like

```jsx
const referralCode = search.match(/referralCode=(\w+)/)[1]
const refSource = search.match(/refSource=(\w+)/)[1]
```

at the moment of submit, unless you think it will change before they submit?

Reply:
Yeah, I suppose I was thinking that someone could remove the param before submitting, but its not a big deal. I agree that it'd be cleaner to not use state at all, and only check on submission. I'll change that.